designer/client/conditions/inline-conditions-definition-values.js

```diff
@@ -49,12 +55,31 @@ function SelectValues (props) {
   )
 }
 
+function customValueComponent (fieldType, operator) {
```

Comment:
I think you're trying to do this:
https://reactjs.org/docs/jsx-in-depth.html#choosing-the-type-at-runtime

this fn returning components should really be it's own component

Reply:
Sort of, but not - as the props also differ between the various things it returns. I've amended to something closer to this though.

Reply:
the props you're passing into the components are derived from fieldType and operator, and at least absoluteDateTimeRenderFunctions all accept the same properties (or prop names). All `InlineConditionsDefinitionValue` knows is that c(C)ustomValueComponent is returning a component